instagram-app/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+//import withAuthenticate from './components/Authentication/withAuthenticate';
+import '../../App.css';
+import SearchBar from '../SearchBar/SearchBar';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+
+class PostPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      comments: {
+        text: '',
+        username: ''
+      }
+     
+    }
+  }
+  
+
+  addComment = (e) => {
+    e.preventDefault();
+    const thisIndex = e.target.getAttribute('index')
+    //console.log(this.state.data[thisIndex].comments)
+    this.state.data[thisIndex].comments.push({ text: this.state.comments.text, username: localStorage.getItem('user')})
+    this.setState({
+      data: [...this.state.data ]
+    })
+  }
+
+  handleChange = e => {
+    //console.log(e.target.value)
+    //this.state.data[0].comments.push({ text:e.target.value, username: "Billy Bob"})
+    this.setState({
+      data: [...this.state.data],
+      comments: {
+        text: e.target.value,
+        username: "BillyBob"
+      }
+    })
+  }
+
+ handleSearch = e => {
+    //console.log(e.target.value)
+    //this.setState({data: [...this.state.data]})
+    //const text = e.target.value.split("").toString()
+    //console.log(text)
+    //const newArray = this.state.data.filter(item => (this.setState({data: item})))
+    const newArray = []
+    this.state.data.filter(function(post) {
+      if (post.username === e.target.value) {
+       newArray.push(post)
```

Comment:
No need to separately filter the items from `this.state.data`; in order to correctly filter the posts by search terms, all you'll have to do is filter directly from `dummyData` and use the `.includes` method to include partial matches, like so:
```
const newArray = dummyData.filter(function(post) {
  // Remember that the .filter method callback should return a Boolean expression that determines what items to include in the filtered array
  return post.username.includes(e.target.value);
});
```